### PR TITLE
multiprocessing gwas pipeline

### PR DESCRIPTION
**Pull request by juhis**

### CODE REVIEW

File: wdl/gwas/saige_sub.multiproc.wdl [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            --numLinesOutput=1000 \
+            --IsOutputAFinCaseCtrl=TRUE \
+            --LOCO=${loco} '
+        for file in '${sep=" " bgenfiles}'.split(' '):
+            cmd = cmd_prefix + '--bgenFile=' + file
+            cmd = cmd + ' --SAIGEOutputFile=${outfileprefix}' + os.path.basename(file) + '.SAIGE.txt'
+            logfile = open('SAIGE_log_${outfileprefix}' + os.path.basename(file) + '.txt', 'w')
+            processes.add(subprocess.Popen(shlex.split(cmd), stdout=logfile))
+        for p in processes:
+            if p.poll() is None:
```
Comment (Fedja) [inline wdl/gwas/saige_sub.multiproc.wdl]: is both poll and wait really needed or could you just wait?

Comment (juhis) [reply wdl/gwas/saige_sub.multiproc.wdl]: Can just wait

File: wdl/gwas/saige_sub.multiproc.wdl [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	output {
+        Array[File] out = glob("*.SAIGE.txt")
+        Array[File] logs = glob("SAIGE_log_*.txt")
+    }
+
+    runtime {
+        docker: "${docker}"
+        cpu: length(bgenfiles)
+        memory: (3.75 * length(bgenfiles)) + " GB"
+        disks: "local-disk " + (length(bgenfiles) * ceil(size(bgenfiles[0], "G")) + 1) + " HDD"
```
Comment (Fedja) [inline wdl/gwas/saige_sub.multiproc.wdl]: Add sort by size suggestion to markdown. Also there could be option to give bgen chunk size as parameter and len(bgens) that amount would be reserved.... that would be just for ease of configuration...anyhoo at least document

File: wdl/gwas/saige_sub.multiproc.wdl (hunk 1 above)
Comment (Fedja) [reply wdl/gwas/saige_sub.multiproc.wdl]: actually... maybe would be better to poll for all jobs and then sleep for a minute and poll again. This way any failed job can be identified and the whole task could throw an error. Now we may be waiting for the first task to finish and some processes might silently fail...

**Pull request closed without merging**